src/groups/heisenberg_group.jl

```diff
@@ -0,0 +1,369 @@
+
+@doc raw"""
+    HeisenbergGroup{T} <: AbstractDecoratorManifold{ℝ}
+
+Heisenberg group `HeisenbergGroup(n)` is the group of ``(n+2)×(n+2)`` matrices [BinzPods:2008](@cite)
+
+```math
+\begin{bmatrix} 1 & \mathbf{a} & c \\
+\mathbf{0} & I_n & \mathbf{b} \\
+0 & \mathbf{0} & 1 \end{bmatrix}
+```
+
+where ``I_n`` is the ``n×n`` unit matrix, ``\mathbf{a}`` is a row vector of length ``n``,
+``\mathbf{b}`` is a column vector of length ``n`` and ``c`` is a real number.
+The group operation is matrix multiplication.
+
+The left-invariant metric on the manifold is used.
+"""
+const HeisenbergGroup{T} = LieGroup{
+    ℝ,MatrixMultiplicationGroupOperation,Manifolds.HeisenbergMatrices{T}
+}
+
+function HeisenbergGroup(n::Int; parameter::Symbol=:type)
+    Hm = Manifolds.HeisenbergMatrices(n; parameter=parameter)
+    return HeisenbergGroup{typeof(Hm).parameters...}(
+        Hm, MatrixMultiplicationGroupOperation()
+    )
+end
+
+function active_traits(f, ::HeisenbergGroup, args...)
```

Comment:
`LieGroup` is not a decorator manifold (on purpose) so, this has zero effect- all `GroupManifold` things are deprecated (soon) anyways.

Yeah, I should write a tutorial after finishing SE

Reply:
Fixed.